### PR TITLE
modifies cmake to link shared lib to the CLI

### PR DESCRIPTION
**Pull request by samhatchett**
also much improved IDE file grouping, for people who use IDEs

### CODE REVIEW

**Comment by samhatchett** [comment]
some final tweaks on the heels of #142 - I'd appreciate any comments from @michaeltryby since he last touched the cmake file

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,18 +56,26 @@ IF (MSVC)
...
-add_executable(runepanet run/main.c $<TARGET_OBJECTS:epanet_objs>)
-if (NOT MSVC)
-    target_link_libraries (runepanet m)    
-endif (NOT MSVC)
+add_executable(runepanet ${EPANET_CLI_SOURCES})
+
+if(NOT MSVC)
+    target_link_libraries(runepanet LINK_PUBLIC epanet m)
+else(NOT MSVC)
+	target_link_libraries(runepanet LINK_PUBLIC epanet)
```
Comment (michaeltryby) [inline CMakeLists.txt]: So under this configuration the `CLE` uses the `libepanet.dll`. Is that correct?

File: CMakeLists.txt [hunk 2]
```diff
@@ -56,18 +56,26 @@ IF (MSVC)
...
 # create object library for reuse in other targets
-include_directories(include; src)
+include_directories(include src)
+
+# configure file groups
 file(GLOB EPANET_SOURCES src/*.c)
-add_library(epanet_objs OBJECT ${EPANET_SOURCES})
+set(EPANET_API_HEADER include/epanet2.h)
+set(EPANET_CLI_SOURCES run/main.c)
+
```
Comment (michaeltryby) [inline CMakeLists.txt]: I had not been using cmake to generate my project files. Good idea!

**Comment by michaeltryby** [comment]
@samhatchett any reservations on merging this PR?

**Comment by samhatchett** [comment]
nope, if it works for you, it also works for me.

**Pull request merged**